### PR TITLE
3846 migrate bulk import pull to batch2

### PR DESCRIPTION
**Pull request by TipzCM**
* migrated bulk import pull to batch2
* added changelog

### CODE REVIEW

File: hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/importpull/BulkImportPullConfig.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			.newBuilder()
+			.setJobDefinitionId(BatchConstants.BULK_IMPORT_JOB_NAME)
+			.setJobDescription("Performs bulk import pull job")
+			.setJobDefinitionVersion(1)
+			.gatedExecution()
+			.setParametersType(Batch2BulkImportPullJobParameters.class)
+			.setParametersValidator(importParameterValidator())
+			.addFirstStep(
+				"ReadInResourcesStep",
+				"Reads an import file and extracs the resources",
```
Comment (nathandoef) [inline hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/importpull/BulkImportPullConfig.java:26]: Nitpick: typo

File: hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/importpull/WriteBundleForImportStep.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+		IBaseResource bundle = parser.parseResource(content);
+
+		// Yeah this is a lame switch - We'll add more later I swear
+		switch (mode) {
+			default:
+			case FHIR_TRANSACTION:
+				IFhirSystemDao systemDao = myDaoRegistry.getSystemDao();
+				systemDao.transaction(requestDetails, bundle);
+//				systemDao.transactionNested(requestDetails, bundle);
```
Comment (nathandoef) [inline hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/importpull/WriteBundleForImportStep.java:70]: Should this line be commented out?

**Review by nathandoef — COMMENTED** [review]
Looks good overall. A couple minor comments

File: hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/importpull/BulkImportPullConfig.java (hunk 1 above)
Comment (TipzCM) [reply hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/importpull/BulkImportPullConfig.java:26]: fixed

File: hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/importpull/WriteBundleForImportStep.java (hunk 2 above)
Comment (TipzCM) [reply hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/importpull/WriteBundleForImportStep.java:70]: removed

File: hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/importpull/WriteBundleForImportStep.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+			jobId,
+			fileIndex
+		);
+
+		IParser parser = myFhirContext.newJsonParser();
+
+		SystemRequestDetails requestDetails = new SystemRequestDetails();
+		requestDetails.setTenantId(tenantName);
+
+		IBaseResource bundle = parser.parseResource(content);
```
Comment (JasonRoberts-smile) [inline hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/importpull/WriteBundleForImportStep.java:64]: question (non-blocking): is there any need to confirm that the resource actually is a Bundle, and not some other type? What would happen if a caller passed in bad data?

**Review by JasonRoberts-smile — APPROVED** [review]
Looks good to me. Approved with a minor question

File: hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/importpull/WriteBundleForImportStep.java (hunk 3 above)
Comment (TipzCM) [reply hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/importpull/WriteBundleForImportStep.java:64]: I don't think so. This code is largely cut-pasted.

We read in the lines from the file (which are assumed to be stringified bundles already) and parse and write them here.

**Pull request merged**